pkg/R/RDD.R

```diff
@@ -1063,7 +1078,7 @@ setMethod("saveAsObjectFile",
           function(rdd, path) {
             # If the RDD is in string format, need to serialize it before saving it because when
             # objectFile() is invoked to load the saved file, only serialized format is assumed.
-            if (!rdd@env$serialized) {
+            if (rdd@env$serialized == "string") {
```

Comment:
Actually this needs to be `rdd@env$serialized != "byte"` because even if the data is serialized as rows we want to convert it to `byte` before saving -- This is because the load code assumes `byte` serialization.